tools/scylla-sstable.cc

```diff
@@ -869,30 +869,52 @@ void consume_sstables(schema_ptr schema, reader_permit permit, std::vector<sstab
     }
 }
 
-using operation_func = void(*)(schema_ptr, reader_permit, const std::vector<sstables::shared_sstable>&, const bpo::variables_map&);
+using read_operation_func = void(*)(schema_ptr, reader_permit, const std::vector<sstables::shared_sstable>&, const bpo::variables_map&);
+using write_operation_func = void(*)(schema_ptr, reader_permit, sstables::sstables_manager&, const bpo::variables_map&);
```

Comment:
I can see operations that do both reads and writes (`compact` operation to compact a bunch of sstables). So how about extending operation_func with sstables_manager? pure writes will have empty input sstables vector, and pure reads will ignore the sstables manager, and a bunch of checks below will disappear.

Reply:
Good idea, will do that.